spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/ClassNLLCriterion.scala

```diff
@@ -93,6 +93,9 @@ class ClassNLLCriterion[@specialized(Float, Double) T: ClassTag]
        s"input dim(${input.dim()})")
     val nClasses = input.size(input.dim())
     if (input.dim() == 1) {
+      if (target.dim() == 2 && target.size(1) == 1) {
+        target.squeeze()
```

Comment:
can we copy to a new tensor? squeeze will change the target' size, sometimes it will lead to some error.

Reply:
Is it right?
```
if (target.dim() == 2 && target.size(1) == 1) {
    val newTarget = target
    newTarget.squeeze()
    val curTarget = ev.toType[Int](newTarget.valueAt(dl = 1))
}
else {
    val curTarget = ev.toType[Int](target.valueAt(dl = 1))
}
```